### PR TITLE
GCW-1487 added the description as package type and multi-line comment

### PR DESCRIPTION
**Pull request by trojanh**

### CODE REVIEW

File: app/components/select-list.js [hunk 1]
```diff
@@ -84,6 +84,8 @@ export default Ember.Component.extend(ValidatableInput, {
         var name = selectedValue.get('name');
         Ember.$("textarea#"+this.get('index')).val(name);
         selectedValue.set('indexOfChild', availablePkg.id);
+
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: Can we remove extra lines?

File: app/components/select-list.js [hunk 2]
```diff
@@ -84,6 +84,8 @@ export default Ember.Component.extend(ValidatableInput, {
...
+
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: ^

File: app/components/select-list.js [hunk 3]
```diff
@@ -84,6 +84,8 @@ export default Ember.Component.extend(ValidatableInput, {
         var name = selectedValue.get('name');
         Ember.$("textarea#"+this.get('index')).val(name);
         selectedValue.set('indexOfChild', availablePkg.id);
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: is this needed?

Comment (trojanh) [inline app/components/select-list.js]: yes, it is used for other select-lists.

Comment (trojanh) [inline app/components/select-list.js]: check https://github.com/crossroads/shared.goodcity/blob/master/app/components/select-list.js

File: app/components/select-list.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
+import Ember from "ember";
+import ValidatableInput from 'ember-cli-html5-validation/mixins/validatable-input';
+
+export default Ember.Component.extend(ValidatableInput, {
+  content: null,
+  selectedValue: null,
+  pkg: null,
+  index: null,
+  flag: true,
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: What is this flag is used for? Can we give a proper name?

File: app/components/select-list.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+  didRender() {
+    this._super(...arguments);
+    if ( this.pkg != null && this.pkg.packageType != null && this.flag === true){
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: In Ember we use convention like **this.get('VariableName')**... Use of **this.variableName** is not preferred.

File: app/components/select-list.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+      if(this.pkg.notes === null || this.pkg.notes.length === 0) {
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: ^

File: app/components/select-list.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        this.pkg.notes = this.pkg.packageType.get('name');
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: ^

File: app/components/select-list.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+      }
+      Ember.$("textarea#"+this.index).val(this.pkg.notes);
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: ^

File: app/components/select-list.js [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+      this.flag=false;
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: ^

File: app/components/select-list.js [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: Extra line

File: app/components/select-list.js [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+  },
+
+  didUpdate() {
+    this._super(...arguments);
+    if ( this.flag === false && this.pkg != null && this.pkg.packageType != null && (this.pkg.notes === null || this.pkg.notes.length === 0 ) ){
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: Follow convention of **this.get("variableName")**

File: app/components/select-list.js [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+      this.flag=true;
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: ^

File: app/components/select-list.js [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+      // this.pkg.notes = this.pkg.packageType.get('name');
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: Remove the comment if not needed

File: app/components/select-list.js [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+       Ember.$("textarea#"+this.index).val(this.pkg.notes);
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: Follow convention of this.get("variableName")

File: app/components/select-list.js [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+
+    if (!this.get('wasValidated')) {
+      jQueryElement.off('focusout').on('keyup', Ember.run.bind(this, this.validate));
+      this.set('wasValidated', true);
+    }
+  },
+
+  actions: {
+    change() {
+      const changeAction  = this.get('on-change');
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: What is the need of declaring const variable here?

File: app/components/select-list.js [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+      const selectedIndex = this.$('select').prop('selectedIndex');
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: ^

File: app/components/select-list.js [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+      var content         = this.get('content').toArray();
+      if (this.get("prompt")) { content = [{name:null}].concat(content); }
+      const selectedValue = content[selectedIndex];
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: ^

File: app/controllers/review_item/accept.js [hunk 18]
```diff
@@ -1,5 +1,5 @@
 import Ember from 'ember';
-import substring from 'goodcity/utils/substring';
+// import substring from 'goodcity/utils/substring';
```
Comment (Darshan-Chauhan) [inline app/controllers/review_item/accept.js]: Remove the comment if not needed

File: app/controllers/review_item/accept.js [hunk 19]
```diff
@@ -62,22 +62,25 @@ export default Ember.Controller.extend({
       itemType.get("defaultChildPackagesList").apply(itemType)
         .forEach(t => this.send("addPackage", t.get("id")));
     }
+
```
Comment (Darshan-Chauhan) [inline app/controllers/review_item/accept.js]: Extra line

File: app/controllers/review_item/accept.js [hunk 20]
```diff
@@ -62,22 +62,25 @@ export default Ember.Controller.extend({
...
+
```
Comment (Darshan-Chauhan) [inline app/controllers/review_item/accept.js]: ^

**Comment by Darshan-Chauhan** [comment]
@trojanh From next time I suggest you refactor your code first and then only submit for PR review. Thanks.

CC: @namrataukirde @steveyken

File: app/components/select-list.js (hunk 4 above)
Comment (trojanh) [inline app/components/select-list.js]: will rename it to make more sense.

File: app/components/select-list.js (hunk 5 above)
Comment (trojanh) [inline app/components/select-list.js]: Will try to find out why we need to use this.get('VariableName') over this.variableName. Is there any specific reason that I am missing here that you are aware of?

File: app/components/select-list.js (hunk 15 above)
Comment (trojanh) [inline app/components/select-list.js]: it is a legacy code used by other select-list as you can see in the shared folder.
I can remove it but its consequences might not be in our favour. 
https://github.com/crossroads/shared.goodcity/blob/master/app/components/select-list.js

File: app/components/select-list.js (hunk 5 above)
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: You always need to use Em.get and Em.set for getting and setting properties of an Ember.Object. It's the basic rule. Without it you may find variety of bugs in observers/rendering and other places.

File: app/components/select-list.js (hunk 15 above)
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: All right. You can keep it as it is

File: app/components/select-list.js (hunk 5 above)
Comment (trojanh) [inline app/components/select-list.js]: i was trying to find out why some variables are accessible from this.variableName and some are not. This is interesting.

File: app/components/select-list.js [hunk 21]
```diff
@@ -0,0 +1,81 @@
...
+export default Ember.Component.extend(ValidatableInput, {
+  content: null,
+  selectedValue: null,
+  pkg: null,
+  index: null,
+  allowValueSet: true,
+
+  didRender() {
+    this._super(...arguments);
+    if ( this.pkg != null && this.pkg.packageType != null && this.allowValueSet === true){
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: this.get

File: app/components/select-list.js [hunk 22]
```diff
@@ -0,0 +1,81 @@
...
+      if(this.pkg.notes === null || this.pkg.notes.length === 0) {
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: ^

File: app/components/select-list.js [hunk 23]
```diff
@@ -0,0 +1,81 @@
...
+        this.pkg.notes = this.pkg.packageType.get('name');
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: ^

File: app/components/select-list.js [hunk 24]
```diff
@@ -0,0 +1,81 @@
...
+      }
+      Ember.$("textarea#"+this.index).val(this.pkg.notes);
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: ^

File: app/components/select-list.js [hunk 25]
```diff
@@ -0,0 +1,81 @@
...
+      this.allowValueSet = false;
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: ^

File: app/components/select-list.js [hunk 26]
```diff
@@ -0,0 +1,81 @@
...
+    }
+  },
+
+  didUpdate() {
+    this._super(...arguments);
+    if ( this.allowValueSet === false && this.pkg != null && this.pkg.packageType != null && (this.pkg.notes === null || this.pkg.notes.length === 0 ) ){
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: ^

File: app/components/select-list.js [hunk 27]
```diff
@@ -0,0 +1,81 @@
...
+      this.allowValueSet=true;
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: ^

File: app/components/select-list.js [hunk 28]
```diff
@@ -0,0 +1,81 @@
...
+      Ember.$("textarea#"+this.index).val(this.pkg.notes);
```
Comment (Darshan-Chauhan) [inline app/components/select-list.js]: ^

**Pull request merged**